addons/payment_razorpay/models/payment_transaction.py

```diff
@@ -20,6 +20,44 @@
 class PaymentTransaction(models.Model):
     _inherit = 'payment.transaction'
 
+    @api.model
+    def _validate_and_sanatize_phone_number(self, phone):
```

Comment:
This could be a payment utils, as it could be used for other providers too at some point, although it looks like for now, none of them is ensuring the number is valid, unlike Razorpay.